src/main/java/io/vertx/core/http/HttpResponseExpectation.java

```diff
@@ -0,0 +1,394 @@
+   */
+  HttpResponseExpectation SC_UPGRADE_REQUIRED = status(426);
+
+  /**
+   * 428 Precondition Required (RFC6585)
+   */
+  HttpResponseExpectation SC_PRECONDITION_REQUIRED = status(428);
+
+  /**
+   * 429 Too Many Requests (RFC6585)
+   */
+  HttpResponseExpectation SC_TOO_MANY_REQUESTS = status(429);
+
+  /**
+   * 431 Request Header Fields Too Large (RFC6585)
+   */
+  HttpResponseExpectation SC_REQUEST_HEADER_FIELDS_TOO_LARGE = status(431);
+
+  /**
+   * Any 5XX server error
+   */
+  HttpResponseExpectation SC_SERVER_ERRORS = status(500, 600);
+
+  /**
+   * 500 Internal Server Error
+   */
+  HttpResponseExpectation SC_INTERNAL_SERVER_ERROR = status(500);
+
+  /**
+   * 501 Not Implemented
+   */
+  HttpResponseExpectation SC_NOT_IMPLEMENTED = status(501);
+
+  /**
+   * 502 Bad Gateway
+   */
+  HttpResponseExpectation SC_BAD_GATEWAY = status(502);
+
+  /**
+   * 503 Service Unavailable
+   */
+  HttpResponseExpectation SC_SERVICE_UNAVAILABLE = status(503);
+
+  /**
+   * 504 Gateway Timeout
+   */
+  HttpResponseExpectation SC_GATEWAY_TIMEOUT = status(504);
+
+  /**
+   * 505 HTTP Version Not Supported
+   */
+  HttpResponseExpectation SC_HTTP_VERSION_NOT_SUPPORTED = status(505);
+
+  /**
+   * 506 Variant Also Negotiates (RFC2295)
+   */
+  HttpResponseExpectation SC_VARIANT_ALSO_NEGOTIATES = status(506);
+
+  /**
+   * 507 Insufficient Storage (WebDAV, RFC4918)
+   */
+  HttpResponseExpectation SC_INSUFFICIENT_STORAGE = status(507);
+
+  /**
+   * 510 Not Extended (RFC2774)
+   */
+  HttpResponseExpectation SC_NOT_EXTENDED = status(510);
+
+  /**
+   * 511 Network Authentication Required (RFC6585)
+   */
+  HttpResponseExpectation SC_NETWORK_AUTHENTICATION_REQUIRED = status(511);
+
+  /**
+   * Creates a predicate asserting that the status response code is equal to {@code statusCode}.
+   *
+   * @param statusCode the expected status code
+   */
+  static HttpResponseExpectation status(int statusCode) {
+    return status(statusCode, statusCode + 1);
+  }
+
+  static HttpResponseExpectation status(int min, int max) {
+    return new HttpResponseExpectation() {
+      @Override
+      public boolean test(HttpClientResponse value) {
+        int sc = value.statusCode();
+        if (sc >= min && sc < max) {
+          return true;
+        }
+        return false;
+      }
+
+      @Override
+      public Exception describe(HttpClientResponse value) {
+        int sc = value.statusCode();
+        if (max - min == 1) {
+          return new VertxException("Response status code " + sc + " is not equal to " + min);
+        }
+        return new VertxException("Response status code " + sc + " is not between " + min + " and " + max);
+      }
+    };
+  }
+
+  /**
+   * Creates a predicate validating the response has a {@code content-type} header matching one of the {@code mimeTypes}.
+   *
+   * @param mimeTypes the list of mime types
+   */
+  static HttpResponseExpectation contentType(List<String> mimeTypes) {
```

Comment:
```suggestion
  static HttpResponseExpectation contentType(String... mimeTypes) {
```
(so we can set 0..n content types)

Reply:
maybe we need both, I expect this code to be also used from WebClient expectations.